### PR TITLE
Add ansible inventory generation

### PR DESCRIPTION
**Pull request by pyaillet**
The goal of this PR is to be able to generate an ansible inventory from various tags.
Example:

```
[pyaillet] # <= User tag
63.32.44.140 tagged=yes private_ip=10.0.7.158 dns_name=ec2-63-32-44-140.eu-west-1.compute.amazonaws.com ansible_ssh_private_key_file=/host/20190213-kubeadmin/users/pyaillet/pyaillet-20190213-kubeadmin.pem

[kubeadm-1] # <= Lab tag
63.32.44.140 tagged=yes private_ip=10.0.7.158 dns_name=ec2-63-32-44-140.eu-west-1.compute.amazonaws.com ansible_ssh_private_key_file=/host/20190213-kubeadmin/users/pyaillet/pyaillet-20190213-kubeadmin.pem

[node-0] # <= node name extracted from the instance name which is: "<user>-<tag>-<count>"
63.32.44.140 tagged=yes private_ip=10.0.7.158 dns_name=ec2-63-32-44-140.eu-west-1.compute.amazonaws.com ansible_ssh_private_key_file=/host/20190213-kubeadmin/users/pyaillet/pyaillet-20190213-kubeadmin.pem
```

### CODE REVIEW

**Comment by pyaillet** [comment]
Also add ssh_config:

```

Host node-0
     HostName 63.32.44.140
     User ubuntu
     IdentityFile /host/20190213-kubeadmin/users/pyaillet/pyaillet-20190213-kubeadmin.pem
```

File: train/train.py [hunk 1]
```diff
@@ -51,6 +51,14 @@
                     help='List running labs and instances in AWS',
                     action='store_true', required=False)
 
+parser.add_argument('-n',
```
Comment (zigarn) [inline train/train.py]: train/train.py:54:26: W291 trailing whitespace

File: train/vpc/labs.py [hunk 2]
```diff
@@ -137,6 +137,71 @@ def get_user_instance_info(conn, user_vpc, lab_tag, user):
     return final
 
 
+def ssh_config_instance(instance, user):
+    if not 'Name' in instance.tags.keys():
```
Comment (zigarn) [inline train/vpc/labs.py]: train/vpc/labs.py:141:8: E713 test for membership should be 'not in'

File: train/vpc/labs.py [hunk 3]
```diff
@@ -137,6 +137,71 @@ def get_user_instance_info(conn, user_vpc, lab_tag, user):
...
+    name = instance.tags['Name']
+    node = "-".join(name.split("-")[1:])
+    return """
+Host {0}
+     HostName {1}
+     User ubuntu
+     IdentityFile /host/{2}/users/{3}/{3}-{2}.pem""".format(node,
+                                                            instance.ip_address,
+                                                            VPC,
+                                                            user)
```
Comment (zigarn) [inline train/vpc/labs.py]: train/vpc/labs.py:152:66: W291 trailing whitespace

File: train/vpc/labs.py [hunk 4]
```diff
@@ -137,6 +137,71 @@ def get_user_instance_info(conn, user_vpc, lab_tag, user):
...
+    return """
+Host {0}
+     HostName {1}
+     User ubuntu
+     IdentityFile /host/{2}/users/{3}/{3}-{2}.pem""".format(node,
+                                                            instance.ip_address,
+                                                            VPC,
+                                                            user)                                                            
+
+def ssh_config(conn, user_vpc, user):
```
Comment (zigarn) [inline train/vpc/labs.py:155]: train/vpc/labs.py:154:1: E302 expected 2 blank lines, found 1

File: train/vpc/labs.py [hunk 5]
```diff
@@ -137,6 +137,71 @@ def get_user_instance_info(conn, user_vpc, lab_tag, user):
...
+    """Generate ssh_config for user"""
+    reservations = conn.get_all_instances(filters = {'vpc-id': user_vpc.id,
```
Comment (zigarn) [inline train/vpc/labs.py]: train/vpc/labs.py:156:50: E251 unexpected spaces around keyword / parameter equals

File: train/vpc/labs.py [hunk 6]
```diff
@@ -137,6 +137,71 @@ def get_user_instance_info(conn, user_vpc, lab_tag, user):
...
+                                                     'tag:User': user})
+    for r in reservations:
+        for instance in r.instances:
+            print ssh_config_instance(instance, user)
+
+def instance_to_inventory(instance):
```
Comment (zigarn) [inline train/vpc/labs.py:164]: train/vpc/labs.py:162:1: E302 expected 2 blank lines, found 1

File: train/vpc/labs.py [hunk 7]
```diff
@@ -137,6 +137,71 @@ def get_user_instance_info(conn, user_vpc, lab_tag, user):
...
+    if 'User' in instance.tags.keys():
+        return "{0} tagged=yes private_ip={1} dns_name={2} \
+ansible_ssh_private_key_file=/host/{3}/users/{4}/{4}-{3}.pem\n".format(instance.ip_address,
+                                           instance.private_ip_address,
```
Comment (zigarn) [inline train/vpc/labs.py]: train/vpc/labs.py:166:44: E128 continuation line under-indented for visual indent
train/vpc/labs.py:167:44: E128 continuation line under-indented for visual indent
train/vpc/labs.py:168:44: E128 continuation line under-indented for visual indent
train/vpc/labs.py:169:44: E128 continuation line under-indented for visual indent

File: train/vpc/labs.py [hunk 8]
```diff
@@ -137,6 +137,71 @@ def get_user_instance_info(conn, user_vpc, lab_tag, user):
...
+                                           instance.public_dns_name,
+                                           VPC,
+                                           instance.tags['User'])
+    else:
+        return ""
+
+def add_to_dict(instance, aux, tag):
```
Comment (zigarn) [inline train/vpc/labs.py:177]: train/vpc/labs.py:173:1: E302 expected 2 blank lines, found 1

File: train/vpc/labs.py [hunk 9]
```diff
@@ -137,6 +137,71 @@ def get_user_instance_info(conn, user_vpc, lab_tag, user):
...
+    if not tag in instance.tags.keys():
```
Comment (zigarn) [inline train/vpc/labs.py]: train/vpc/labs.py:174:8: E713 test for membership should be 'not in'

File: train/vpc/labs.py [hunk 10]
```diff
@@ -137,6 +137,71 @@ def get_user_instance_info(conn, user_vpc, lab_tag, user):
...
+        return
+    if not instance.tags[tag] in aux.keys():
+        aux[instance.tags[tag]] = ""
+    aux[instance.tags[tag]] += instance_to_inventory(instance)
+
+def add_node_to_instance(instance):
```
Comment (zigarn) [inline train/vpc/labs.py:184]: train/vpc/labs.py:180:1: E302 expected 2 blank lines, found 1

File: train/vpc/labs.py [hunk 11]
```diff
@@ -137,6 +137,71 @@ def get_user_instance_info(conn, user_vpc, lab_tag, user):
...
+    if not 'Name' in instance.tags.keys():
```
Comment (zigarn) [inline train/vpc/labs.py]: train/vpc/labs.py:181:8: E713 test for membership should be 'not in'

File: train/vpc/labs.py [hunk 12]
```diff
@@ -137,6 +137,71 @@ def get_user_instance_info(conn, user_vpc, lab_tag, user):
...
+        return
+    name = instance.tags['Name']
+    node = "-".join(name.split("-")[1:])
+    instance.tags['Node'] = node
+
+def inventory(conn, user_vpc):
```
Comment (zigarn) [inline train/vpc/labs.py:192]: train/vpc/labs.py:187:1: E302 expected 2 blank lines, found 1

File: train/vpc/labs.py [hunk 13]
```diff
@@ -137,6 +137,71 @@ def get_user_instance_info(conn, user_vpc, lab_tag, user):
...
+    users = {}
+    nodes = {}
+    labs = {}
+    reservations = conn.get_all_instances(filters = {'vpc-id': user_vpc.id})
```
Comment (zigarn) [inline train/vpc/labs.py]: train/vpc/labs.py:191:50: E251 unexpected spaces around keyword / parameter equals

File: train/vpc/labs.py [hunk 14]
```diff
@@ -137,6 +137,71 @@ def get_user_instance_info(conn, user_vpc, lab_tag, user):
...
+            add_to_dict(instance, labs, 'Lab')
+            add_node_to_instance(instance)
+            add_to_dict(instance, nodes, 'Node')
+    for k, v in users.iteritems():
+        print("[{0}]\n{1}".format(k, v))
+    for k, v in labs.iteritems():
+        print("[{0}]\n{1}".format(k, v))
+    for k, v in nodes.iteritems():
+        print("[{0}]\n{1}".format(k, v))
+
```
Comment (zigarn) [inline train/vpc/labs.py:209]: train/vpc/labs.py:205:1: E302 expected 2 blank lines, found 1

**Review by zigarn — CHANGES_REQUESTED** [review]
Erreurs PEP8 (exécution de `pycodestyle`)

File: train/vpc/labs.py [hunk 15]
```diff
@@ -137,6 +137,71 @@ def get_user_instance_info(conn, user_vpc, lab_tag, user):
     return final
 
 
+def ssh_config_instance(instance, user):
+    if not 'Name' in instance.tags.keys():
+        return ""
+    name = instance.tags['Name']
+    node = "-".join(name.split("-")[1:])
+    return """
```
Comment (zigarn) [inline train/vpc/labs.py]: Serait plus lisible avec le template dans une variable puis le `format` sur ce template je pense

File: train/vpc/labs.py [hunk 16]
```diff
@@ -137,6 +137,71 @@ def get_user_instance_info(conn, user_vpc, lab_tag, user):
...
+    """Generate ssh_config for user"""
+    reservations = conn.get_all_instances(filters = {'vpc-id': user_vpc.id,
+                                                     'tag:User': user})
+    for r in reservations:
+        for instance in r.instances:
+            print ssh_config_instance(instance, user)
+
+def instance_to_inventory(instance):
+    if 'User' in instance.tags.keys():
+        return "{0} tagged=yes private_ip={1} dns_name={2} \
```
Comment (zigarn) [inline train/vpc/labs.py]: Serait plus lisible avec le template dans une variable puis le `format` sur ce template je pense

File: train/vpc/labs.py [hunk 17]
```diff
@@ -137,6 +137,71 @@ def get_user_instance_info(conn, user_vpc, lab_tag, user):
...
+                                           instance.public_dns_name,
+                                           VPC,
+                                           instance.tags['User'])
+    else:
+        return ""
+
+def add_to_dict(instance, aux, tag):
+    if not tag in instance.tags.keys():
+        return
+    if not instance.tags[tag] in aux.keys():
```
Comment (zigarn) [inline train/vpc/labs.py]: A priori, pas besoin du `.keys()`

File: train/vpc/labs.py [hunk 18]
```diff
@@ -137,6 +137,71 @@ def get_user_instance_info(conn, user_vpc, lab_tag, user):
...
+        aux[instance.tags[tag]] = ""
+    aux[instance.tags[tag]] += instance_to_inventory(instance)
```
Comment (zigarn) [inline train/vpc/labs.py]: Ou sinon ça pour les 3 précédentes lignes :

```python
aux[instance.tags[tag]] = aux.get(instance.tags[tag], "") + instance_to_inventory(instance)
```

**Review by zigarn — CHANGES_REQUESTED** [review]

File: train/vpc/labs.py (hunk 15 above)
Comment (pyaillet) [reply train/vpc/labs.py]: 👍
Je l'ai fait pour d'autres également

File: train/vpc/labs.py (hunk 16 above)
Comment (pyaillet) [reply train/vpc/labs.py]: 👍

**Comment by pyaillet** [comment]
Thanks for the review, I made some fixes !

**Review by zigarn — APPROVED** [review]

**Pull request merged**